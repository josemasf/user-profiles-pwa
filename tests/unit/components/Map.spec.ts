import {
    shallowMount
} from '@vue/test-utils'
import {
    Map
} from '@/components'

describe('Map.vue', () => {
    const wrapper: any = shallowMount(Map, {
        propsData: {
            latitude: 0,
            longitude: 0,
            id: 1
        }
      });

    describe('Rendering', () => {
        it('Se renderiza el componente', async () => {
            expect(wrapper).toMatchSnapshot();
        });
    })
    describe('Props', () => {
        it('latitude exists', async () => {
          const prop = wrapper.props().latitude;
          expect(prop).toBe(0);
        });
    
        it('longitude exists', async () => {
          const prop = wrapper.props().longitude;
          expect(prop).toBe(0);
        });
        it('id exists', async () => {
          const prop = wrapper.props().id;
          expect(prop).toBe(1);
        });   

      });
})