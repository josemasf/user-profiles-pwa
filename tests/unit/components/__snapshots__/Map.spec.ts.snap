// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map.vue Rendering Se renderiza el componente 1`] = `
<div style="height: 200px; width: 95%;">
  <l-map-stub options="[object Object]" center="LatLng(0, 0)" zoom="10" crs="[object Object]">
    <l-tile-layer-stub pane="tilePane" attribution="&amp;copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors" visible="true" opacity="1" zindex="1" tilesize="256" subdomains="abc" options="[object Object]" url="http://{s}.tile.osm.org/{z}/{x}/{y}.png" tilelayerclass="[Function]"></l-tile-layer-stub>
    <l-marker-stub pane="markerPane" visible="true" options="[object Object]" latlng="LatLng(0, 0)" icon="[object Object]" opacity="1"></l-marker-stub>
  </l-map-stub>
</div>
`;
